src/context/theme-provider.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+
+export interface ThemeProviderProps {
+  children: React.ReactNode
+}
```

Comment:
instead of explicitly setting children for props, you can `extend React.FC<PropsWithChildren>` or similar.. 

or, my preference since there are no custom props, remove this and do it below

```suggestion
```

Reply:
I think there's this minor issue with using `React.FC` in newer versions of React and I understand that it sits well with this component since it expects children which `React.FC<PropsWithChildren>` provides implicitly. 

why shouldn't we set the `children` prop explicitly?

Reply:
@SgtPooki

Reply:
@kaf-lamed-beyt no huge reason, but here are a few: 

* prevent issues if React changes the type of the children prop
* less code to maintain
* using prop types from react directly prevents needing to add other props if we need to extend them (adding className or other common props in the future)

Reply:
> I think there's this minor issue with using React.FC in newer versions of React

I haven't ran into this, can you link to examples?

Reply:
huh. makes sense. 

for this 👇🏼 
> using prop types from react directly prevents needing to add other props if we need to extend them (adding className or other common props in the future)

I may be wrong... but, can't we do `...props`?

> I haven't ran into this, can you link to examples?

This [issue](https://github.com/facebook/create-react-app/pull/8177) from facebook/create-react-app sheds more light on why they deprecated `React.FC` @SgtPooki